tests/core/test_encode_structured_data_vs_signTypedData_v4.py

```diff
@@ -0,0 +1,503 @@
+from eth_utils.exceptions import (
+    ValidationError,
+)
+import pytest
+
+from eth_account.messages import (
+    _hash_eip191_message,
+    encode_structured_data,
+)
+
+"""
+`encode_structured_data` is eth-account's take on EIP-712 typed structured data hashing.
+
+signTypedData is MetaMask's interpretation:
+https://github.com/MetaMask/eth-sig-util/tree/main
+
+Their signTypedData_v4 has become a de facto standard.
+
+There is no clarification in the EIP about how to handle missing or extra data.
+signTypedData_v4 has taken a more permissive approach than eth-account has.
+
+The tests below will clarify what encode_structured_data allows.
+The table and tests borrow heavily from `eth-sig-util`'s:
+https://github.com/MetaMask/eth-sig-util/blob/main/src/sign-typed-data.test.ts
+
+
+| Input type                                           | esd |
```

Comment:
nit: I had to look a few times to figure out what `esd` was. It might be clearer to just type it out 🤷

Reply:
typed out